src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -161,24 +167,75 @@ impl StreamingCommandRunner {
       .await
   }
 
-  async fn ensure_action_uploaded(
+  async fn check_action_cache(
```

Comment:
A method doc that includes some of the explanation from that thread to explain why we do this manually would be great.

Reply:
Will add.

Reply:
Added.